packages/grafana-data/src/rbac/rbac.ts

```diff
@@ -0,0 +1,30 @@
+import { CurrentUserDTO } from '../types';
+
+export interface CurrentUser extends Omit<CurrentUserDTO, 'lightTheme'> {}
+
+export function userHasPermission(action: string, user: CurrentUser): boolean {
+  return !!user.permissions?.[action];
+}
+
+export function userHasPermissionInMetadata(
+  action: string,
+  object: {
+    accessControl?: Record<string, boolean>;
+  }
+): boolean {
+  return !!object.accessControl?.[action];
+}
+
+export function userHasAllPermissions(actions: string[], user: CurrentUser) {
+  if (actions.every((action) => userHasPermission(action, user))) {
+    return [];
+  }
+  return ['Reject'];
+}
```

Comment:
Should we have a better return type here instead of returning a string array?

Reply:
Yes, I wanted to return a boolean but then I also wanted a single source of truth so replaced the code in the contextSrv methods to use these functions. Maybe I'm missing something but this means that changing the return type here would cause a breaking change for any plugin that currently uses `contextSrv.hasPermission` etc?

Perhaps one option would be to change the signature of the functions exposed via runtime to return booleans based on the result of these functions? WDYT @mckn?

Reply:
Have updated them based on our conversation to return booleans then used them in contextSrv in a non-breaking way.